pkg/pipelines/bootstrap_test.go

```diff
@@ -158,3 +159,38 @@ func TestApplicationFromRepo(t *testing.T) {
 		t.Fatalf("bootstrapped resources:\n%s", diff)
 	}
 }
+
+func TestOverwriteFlag(t *testing.T) {
+	defer func(f secrets.PublicKeyFunc) {
+		secrets.DefaultPublicKeyFunc = f
+	}(secrets.DefaultPublicKeyFunc)
+
+	secrets.DefaultPublicKeyFunc = func(service types.NamespacedName) (*rsa.PublicKey, error) {
+		key, err := rsa.GenerateKey(rand.Reader, 1024)
+		if err != nil {
+			t.Fatalf("failed to generate a private RSA key: %s", err)
+		}
+		return &key.PublicKey, nil
+	}
+	fakeFs := ioutils.NewMapFilesystem()
+	params := &BootstrapOptions{
+		InitOptions: &InitOptions{
+			Prefix:              "tst-",
+			GitOpsRepoURL:       testGitOpsRepo,
+			ImageRepo:           "image/repo",
+			GitOpsWebhookSecret: "123",
+		},
+		ServiceRepoURL:       testSvcRepo,
+		ServiceWebhookSecret: "456",
+	}
+	err := Bootstrap(params, fakeFs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got := Bootstrap(params, fakeFs)
+	want := fmt.Errorf("Directory already exists, cannot overwrite ( set --overwrite=true to continue )")
```

Comment:
Why use `Errorf` here? there are no format strings?

Reply:
We now use a single function, for both init and bootstrap.
The command line descriptions were capitalized.
The fmt.Errorf was changed since no formatting occurs.
Description of error message now more clear